core/bin/oauth.rs

```diff
@@ -0,0 +1,222 @@
+use anyhow::anyhow;
+use axum::{
+    extract::{Path, State},
+    response::Json,
+    routing::{get, post},
+    Router,
+};
+use dust::{
+    oauth::{
+        connection::{Connection, ConnectionProvider},
+        store,
+    },
+    utils::{error_response, APIResponse, CoreRequestMakeSpan},
+};
+use hyper::StatusCode;
+use serde::Deserialize;
+use serde_json::json;
+use std::sync::Arc;
+use tokio::{
+    net::TcpListener,
+    signal::unix::{signal, SignalKind},
+};
+use tower_http::trace::{self, TraceLayer};
+use tracing::{error, info, Level};
+use tracing_bunyan_formatter::{BunyanFormattingLayer, JsonStorageLayer};
+use tracing_subscriber::prelude::*;
+
+struct OAuthState {
+    store: Box<dyn store::OAuthStore + Sync + Send>,
+}
+
+impl OAuthState {
+    fn new(store: Box<dyn store::OAuthStore + Sync + Send>) -> Self {
+        Self { store }
+    }
+}
+
+async fn index() -> &'static str {
+    "oauth server ready"
+}
+
+#[derive(Deserialize)]
+struct ConnectionCreatePayload {
+    provider: ConnectionProvider,
+    metadata: serde_json::Value,
+}
+
+async fn connections_create(
```

Comment:
Can we move everything related to connection/DB handling to a dedicated place, outside of the main file that host the API endpoints?

Reply:
wdym by connection/db handling?

Reply:
I meant these two functions, not sure what's the best Rust web app pattern but I found it weird that it lives alongside the endpoints definition.  It's a nit though.

Reply:
Well this is what we do in other services. These are the endpoint definitions and we currently put them for now in the bin file along with the Router definition :+1: